### ISSUE TITLE
Make it possible to render Promise[VDomModifier] and Future[VDomModifier]

### ISSUE THREAD

**Issue by fdietze** [OutWatch/outwatch#256]
I just had this idea. It would quite convenient, since one can immediately use results of e.g. http-requests:

```scala
myRequest:Future[String] = ???
div("request result: ", myRequest)
```

**Comment by mariusmuja** [comment]
Easily accomplished by adding this to `AsVDomModifier`:

```scala
implicit def taskLikeRender[T, F[_]: TaskLike](implicit r: AsVDomModifier[T]): AsVDomModifier[F[T]] = (value: F[T]) =>
    Observable.fromTaskLike(value)
```

Then it will work with `Future`, but also cats-effect `IO`, `SyncIO`, `Eval`, `Effect`, `ConcurrentEffect`, monix `Task`, `Coeval`, scala `Try`, `Function0`, `Either[T, Throwable]`.

**Comment by cornerman** [comment]
Nice, sounds like a good idea!

**Comment by cornerman** [comment]
We just need to see how we go about IO, currently we have this a type for managing sideeffects in our API. But you do want these sideeffect to be executed synchronously when patching. With a `AsVDomModifier[TaskLike]`, we would have two conflicting implicit for IO. This might work but it might not be intuitive. What is the best approach here to differentiate?

**Comment by cornerman** [comment]
To add to this: I think what we really need is a way to differentiate three effects that we can have as a modifier.

1. A `TaskLike` or `Effect` or specifically `IO`:  An example would be a network request that you want to fire when inserting the dom. So, we have a sideeffect that is to be executed in an async fashion.
2. A synchronous effect that you have in your application: An example would be `Handler.create`, it is a sideeffect in the sense that it carries state and exposes and `Observer` interface to mutate the state. If you want to build referentially transparent element, this is the way to go. You can still use the same state by mapping over the effect and using the inner handler in mutilple parts of your page. But these effect should really be done *sync*, otherwise you will have a whole bunch of snabbdom patches when rendering.
3. A synchronous effect that should never ever be included into more than one element: An example is `managed`. This tracks the element for one single VNode instance that is inserted into the dom. It currently returns an `IO`, but it should never be mapped over and be reused. You would want that to be a `VDomModifier.effect { .... }`, so you can never mistakenly use it in two elements at the same time.

**Comment by cornerman** [comment]
Fixed in #355
